Tasks/CopyPublishBuildArtifacts/CopyPublishBuildArtifacts.ps1

```diff
@@ -28,6 +28,13 @@ Write-Verbose "TargetPath = $TargetPath"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 
+$hostType = Get-TaskVariable $distributedTaskContext "system.hosttype"
+
+if($hostType -ine "build" )
```

Comment:
This check and message reads task to be more appropriate to build, instead of saying where this task is not supported, I feel it's better to say where it is supported in case user sees this error..